File: lab3/src/main/java/lab3/myquest/MyQuest.java
package lab3.myquest;

import lab3.functionality.Action;
import lab3.interaction.ConsoleDesigner;
import lab3.functionality.Player;
import lab3.interaction.ConsoleReader;
import lab3.interaction.ConsoleWriter;
import lab3.functionality.Quest;
import lab3.myquest.items.Chair;
import lab3.myquest.locations.Ceil;
import lab3.myquest.locations.Floor;

public class MyQuest extends Quest {
    public MyQuest(Player player) {
        super(new ConsoleDesigner(new ConsoleReader(), new ConsoleWriter(), player), player);
        var ceil = new Ceil("Потолок");
        player.getLocation().addAction(new Action("Выпрямиться", p -> p.getName() + " хотел выпрямиться, но как раз в это время его снизу словно толкнуло что-то и подбросило под потолок.", p -> p.setLocation(ceil)));
        var floor = new Floor("Пол");
        floor.addItem(new Chair("Стул"));
        ceil.addLocation(floor);
    }
}

File: lab4/src/main/kotlin/lab4/myQuest/MyQuest.kt
package lab4.myQuest

import lab4.functionality.Designer
import lab4.functionality.Player
import lab4.functionality.Quest
import lab4.myQuest.locations.Floor
import lab4.myQuest.locations.Space
import lab4.myquest.locations.Ceil

class MyQuest(designer: Designer, player: Player) : Quest(designer, player) {
    init {
        player.location = Space()
        player.location
            .addOneWayLocation(Floor())
            .addOneWayLocation(Ceil())
    }

}